### PR TITLE
Frontend view for concept uploader

### PR DESCRIPTION
**Pull request by Ommy**
Only displays on right navbar when user is a superuser

### CODE REVIEW

**Comment by Kapin** [comment]
Thing to keep in mind, the check for whether to display it or not should validate with the server every single time you navigate to a page that could have it. It should be impossible to reach without superuser status.

**Comment by Ommy** [comment]
@Kapin Realistically if they go into the console and somehow change the user stored in session store, then the backend should still validate the user's status (i.e. in the concept uploader ViewSet so we still get a 401). We could, in the conceptsRouter, create a user model from the user information stored in session store and fetch the model from the backend, and then check on the returned fields.

**Comment by Ommy** [comment]
image

File: src-backbone/app/js/controllers/conceptsController.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const ConceptsUploaderView = require('views/concepts/conceptsUploaderView');
+
+module.exports = Marionette.Controller.extend({
+
+    routeUploader: function () {
+        if (!App().session.isValid()) {
+            Helpers.navigateToDefaultLoggedOut();
+            return;
+        }
+        App().session.refreshUser(false);
```
Comment (Trinovantes) [inline src-backbone/app/js/controllers/conceptsController.js]: You're freezing the UI thread here. Why not just let the async callback redirect the user instead?

**Comment by Trinovantes** [comment]
Some of the indentation is off

If we had the time, I'd also like to rewrite/restructure how we handle users/session models since they're so non-coherent (e.g. instantiating new user model from session attributes?) but still strongly coupled together right now.

**Comment by Ommy** [comment]
Could move all the functions to User model and put accessor in App()

**Comment by Kapin** [comment]
lgtm. I'd make a ticket to uncouple the user and session objects.

**Comment by Ommy** [comment]
Yeah I'll be doing that, or attempting to do that now

**Pull request merged**